File: java/InkRecognition/android-sample-app/Recognizer/app/src/main/java/CognitiveServices/Ink/Recognition/InkLine.java
package CognitiveServices.Ink.Recognition;

import android.util.DisplayMetrics;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import java.util.ArrayList;

class InkLine extends InkRecognitionUnit {

    private float indentLevel = 0.0f;
    private String text;
    private final ArrayList<String> alternates = new ArrayList<>();

    public float getIndentLevel() {
        return indentLevel;
    }

    public String getText() {
        return text;
    }

    public ArrayList<String> getAlternates() {
        return alternates;
    }

    InkLine(String lineJSON, DisplayMetrics metrics, InkRoot result) throws JSONException {
        super(lineJSON, metrics, result);
        JSONObject jsonLine = new JSONObject(lineJSON);
        JSONArray jsonAlternates = jsonLine.getJSONArray("alternates");
        if (jsonLine.has("indentLevel")) {
            this.indentLevel = jsonLine.getInt("indentLevel");
        }
        this.text = jsonLine.getString("recognizedText");

        for (int i=0;
             i < jsonAlternates.length();
             i++) {
            alternates.add(jsonAlternates.getJSONObject(i).getString("recognizedString"));

        }

    }

}

File: swift/InkRecognition/Recognizer/Recognizer/InkLine.swift

import Foundation
@objc
class InkLine : InkRecognitionUnit {
    
    var alternates = [[String:Any]]()
    var indentLevel = 0.0
    var text:String!
    
    @objc
    override init(json : [String: Any])
    {
        self.text = json["recognizedText"] as? String ?? ""
        if let alternates = json["alternates"] as? [[String: Any]] {
            self.alternates = alternates
        }
        super.init(json:json)
    }
    
    @objc
    func getAlternates() -> [String]
    {
        var wordAlternates = [String]()
        for jsonAlternate in self.alternates {
            let wordAlternate = jsonAlternate["recognizedString"] as? String ?? ""
            wordAlternates.append(wordAlternate)
        }
        return wordAlternates
    }
}
